### DISCUSSION TITLE
How can map database table fields with data objects ??

### DISCUSSION THREAD

**Discussion by Kistlak** [spatie/laravel-data#49] [Q&A] [answered]
I have created a Data object called InvoiceData like below. 

```
<?php

class InvoiceData extends Data
{
    #[Rule(['required', 'date'])]
    public string $invoiceDate;

    #[Rule(['required', 'date'])]
    public string $dueDate;

    #[Rule(['nullable', 'string'])]
    public ?string $notes;

    public static function fromModel(Invoice $invoice): self
    {
        return self::from([           
            'invoiceDate' => $invoice->invoice_date,
            'dueDate' => $invoice->due_date,
            'notes' => $invoice->notes,
        ]);
    }
}
```

And I have a create function like below.

```
$invoice = Invoice::create([
            'invoice_date' => $invoiceData->invoiceDate,
            'due_date' => $invoiceData->dueDate,
            'notes' => $invoiceData->notes,
        ]);
```

The thing is like I use camel cases in the properties and use snake cases in the database table fields. How can I map these table fields and data objects ?? I just want to use one variable inside create() method. I use fromModel() and fromRequest() methods. But, I couldn't do this using these 2. 
How can I do this ??

**Comment by rubenvanassche** [comment, marked as answer]
There isn't a proper solution to fix this, in your case, I would create a trait with a method `toEloquentAttributes` here you can transform a data object `toArray` and then replace the keys of the array with snakeCase versions of your camelCase keys.

At Spatie we're using snake case properties within our data objects because it clearly communicates to a developer that a property is part of a data object but that's off course personal preference.
